FWCore/Framework/src/stream/ProducingModuleAdaptorBase.cc

```diff
@@ -179,43 +180,44 @@ namespace edm {
 
     template <typename T>
     void ProducingModuleAdaptorBase<T>::doStreamBeginRun(StreamID id,
-                                                         RunPrincipal const& rp,
-                                                         EventSetupImpl const& ci,
+                                                         RunTransitionInfo const& info,
                                                          ModuleCallingContext const* mcc) {
+      RunPrincipal const& rp = info.principal();
       auto mod = m_streamModules[id];
       setupRun(mod, rp.index());
 
       Run r(rp, moduleDescription_, mcc, false);
       r.setConsumer(mod);
       const EventSetup c{
-          ci, static_cast<unsigned int>(Transition::BeginRun), mod->esGetTokenIndices(Transition::BeginRun), false};
+          info, static_cast<unsigned int>(Transition::BeginRun), mod->esGetTokenIndices(Transition::BeginRun), false};
       mod->beginRun(r, c);
     }
+
     template <typename T>
     void ProducingModuleAdaptorBase<T>::doStreamEndRun(StreamID id,
-                                                       RunPrincipal const& rp,
-                                                       EventSetupImpl const& ci,
+                                                       RunTransitionInfo const& info,
                                                        ModuleCallingContext const* mcc) {
+      RunPrincipal const& rp = info.principal();
```

Comment:
I suppose this line could be omitted, and two lines below the `Run` be constructed with
```cpp
      Run r(info, moduleDescription_, mcc, true);
```

Reply:
Done. Thanks. (I'll push the new commit in a few minutes)